src/python/autotransform/transformer/openai.py

```diff
@@ -58,19 +68,48 @@ def _transform_item(self, item: Item) -> None:
         # Set up messages for prompt
         if self.system_message:
             messages.append({"role": "system", "content": self.system_message})
-        messages.append({
-            "role": "user",
-            "content": self._replace_sentinel_values(self.prompt, item),
-        })
+        messages.append(
+            {
+                "role": "user",
+                "content": self._replace_sentinel_values(self.prompt, item),
+            }
+        )
         # Get completition
         chat_completion = openai.ChatCompletion.create(
             model=self.model,
             messages=messages,
-            temperature=self.temperature
+            temperature=self.temperature,
         )
-        item.write_content(self._extract_code_from_completion(
-            chat_completion.choices[0].message.content,
-        ))
+        result = chat_completion.choices[0].message.content
```

Comment:
Very cool pattern - so it's wired into the transformer itself? Would there be a way to specify a validator external to this transformer?

Reply:
The Validators that are used are supplied by a param that is given to the OpenAITransformer. Any Validator you could make with AutoTransform can be used here. If you take a look at the nathro_test branch and look at the crawler.json schema you'll see how I supplied Validators to it.